src/test/java/edu/harvard/iq/dataverse/util/cache/RateLimitUtilTest.java

```diff
@@ -0,0 +1,128 @@
+import edu.harvard.iq.dataverse.util.cache.RateLimitUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class RateLimitUtilTest {
+
+    private SystemConfig mockedSystemConfig;
+
+    static final String settingJson = "{\n" +
+            "  \"rateLimits\":[\n" +
+            "    {\n" +
+            "      \"tier\": 0,\n" +
+            "      \"limitPerHour\": 10,\n" +
+            "      \"actions\": [\n" +
+            "        \"GetLatestPublishedDatasetVersionCommand\",\n" +
+            "        \"GetPrivateUrlCommand\",\n" +
+            "        \"GetDatasetCommand\",\n" +
+            "        \"GetLatestAccessibleDatasetVersionCommand\"\n" +
+            "      ]\n" +
+            "    },\n" +
+            "    {\n" +
+            "      \"tier\": 0,\n" +
+            "      \"limitPerHour\": 1,\n" +
+            "      \"actions\": [\n" +
+            "        \"CreateGuestbookResponseCommand\",\n" +
+            "        \"UpdateDatasetVersionCommand\",\n" +
+            "        \"DestroyDatasetCommand\",\n" +
+            "        \"DeleteDataFileCommand\",\n" +
+            "        \"FinalizeDatasetPublicationCommand\",\n" +
+            "        \"PublishDatasetCommand\"\n" +
+            "      ]\n" +
+            "    },\n" +
+            "    {\n" +
+            "      \"tier\": 1,\n" +
+            "      \"limitPerHour\": 30,\n" +
+            "      \"actions\": [\n" +
+            "        \"CreateGuestbookResponseCommand\",\n" +
+            "        \"GetLatestPublishedDatasetVersionCommand\",\n" +
+            "        \"GetPrivateUrlCommand\",\n" +
+            "        \"GetDatasetCommand\",\n" +
+            "        \"GetLatestAccessibleDatasetVersionCommand\",\n" +
+            "        \"UpdateDatasetVersionCommand\",\n" +
+            "        \"DestroyDatasetCommand\",\n" +
+            "        \"DeleteDataFileCommand\",\n" +
+            "        \"FinalizeDatasetPublicationCommand\",\n" +
+            "        \"PublishDatasetCommand\"\n" +
+            "      ]\n" +
+            "    }\n" +
+            "  ]\n" +
+            "}";
+    static final String settingJsonBad = "{\n";
+
+    @BeforeEach
+    public void setup() {
+        mockedSystemConfig = mock(SystemConfig.class);
+        doReturn("100,200").when(mockedSystemConfig).getRateLimitingDefaultCapacityTiers();
+        // clear the static data so it can be reloaded with the new mocked data
+        RateLimitUtil.rateLimitMap.clear();
+        RateLimitUtil.rateLimits.clear();
+    }
+    @Test
+    public void testConfig() {
```

Comment:
**suggestion**: This is a perfect candidate for a `@ParameterizedTest`.

Reply:
created ParameterizedTest